packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+    }
+
+    if (prevProps.open !== open) {
+      return open ? this.foundation.open() : this.foundation.close();
+    }
+  }
+
+  get classes(): string {
+    const {classList} = this.state;
+    const {className} = this.props;
+    return classnames(cssClasses.BASE, Array.from(classList), className);
+  }
+
+  get buttons_(): HTMLButtonElement[] | null {
+    return this.dialogElement.current &&
+       [].slice.call(this.dialogElement.current.getElementsByClassName(cssClasses.BUTTON));
+  }
+
+  get content_(): HTMLElement | null {
+    return this.dialogElement.current &&
+        this.dialogElement.current.querySelector(`.${cssClasses.CONTENT}`);
+  }
+
+  get defaultButton_(): HTMLButtonElement | null {
+    return this.dialogElement.current &&
+        this.dialogElement.current.querySelector(`.${cssClasses.DEFAULT_BUTTON}`);
+  }
+
+  private initializeFocusTrap = () => {
+    this.focusTrap = util.createFocusTrapInstance(this.dialogElement.current);
+  }
+
+  get adapter(): Partial<MDCDialogAdapter> {
+    const strings = MDCDialogFoundation.strings;
+    const {closest, matches} = ponyfill;
+    const {isScrollable, areTopsMisaligned} = util;
+    return {
+      addClass: (className: string) => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className: string) => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      addBodyClass: (className: string) => document.body.classList.add(className),
+      removeBodyClass: (className: string) => document.body.classList.remove(className),
+      eventTargetMatches: (target: HTMLElement, selector: string) => matches(target, selector),
+      trapFocus: () => this.focusTrap && this.focusTrap.activate(),
+      releaseFocus: () => this.focusTrap && this.focusTrap.deactivate(),
+      isContentScrollable: () => {
+        const content = this.content_;
+        return (!!content) ? isScrollable(content) : false;
+      },
+      areButtonsStacked: () => areTopsMisaligned(this.buttons_),
+      getActionFromEvent: (evt: any) => {
+        const elem = closest(evt.target, `[${strings.ACTION_ATTRIBUTE}]`);
+        return elem && elem.getAttribute(strings.ACTION_ATTRIBUTE);
+      },
+      clickDefaultButton: () => {
+        const defaultButton = this.defaultButton_;
+        if (defaultButton) {
+          defaultButton.click();
+        }
+      },
+      reverseButtons: () => {
+        const buttons = this.buttons_;
+        return buttons &&
+          buttons.reverse().forEach((button: HTMLButtonElement) =>
+            button.parentElement && button.parentElement.appendChild(button)
+          );
+      },
+      notifyOpening: () => this.handleOpening(),
+      notifyOpened: () => this.props.onOpen && this.props.onOpen(),
```

Comment:
if you add defaults you don't need this && check. Ignore my previous comment if you want to keep it like this.

Reply:
Yes, that makes sense. However, see my previous comment on the defaults here [[0](https://github.com/material-components/material-components-web-react/pull/611#discussion_r248844978)]

[0] https://github.com/material-components/material-components-web-react/pull/611#discussion_r248844978